WcaOnRails/spec/controllers/competitions_controller_spec.rb

```diff
@@ -521,19 +521,19 @@
         expect(competition.reload.registration_close).to eq new_close
       end
 
-      it "can change registration requirements field before competition is confirmed" do
+      it "can change extra registration requirements field before competition is confirmed" do
         new_requirements = "New requirements"
         patch :update, params: { id: competition, competition: { registration_requirements: new_requirements } }
         competition.reload
         expect(competition.registration_requirements).to eq new_requirements
       end
 
-      it "cannot change registration requirements field after competition is confirmed" do
+      it "cannot change extra registration requirements field after competition is confirmed" do
         comp = FactoryBot.create(:competition, :confirmed, delegates: [delegate])
         new_requirements = "New requirements"
         patch :update, params: { id: comp, competition: { registration_requirements: new_requirements } }
         comp.reload
-        expect(comp.registration_requirements).to eq "Requirements"
+        expect(comp.registration_requirements).to be_nil
```

Comment:
This looks problematic. Why did you change this? This test is trying to make sure that you cannot change this field after a competition is confirmed.

Reply:
Now the `registration_requirements` field is not mandatory (this field is for extra registration requirements now), so I removed it from the factory. I believe this test is still testing that the field can't be changed after a competition is confirmed, just that the initial value is different. Let me know if I should change it back, and the factory too.

Reply:
Ohhhh, got it! You're totally right. This test is also a little bit confusing because it depends on the reader knowing that the factory defaults to creating a competition with `nil` registration_requirements. I would rewrite this test as follows:

```
        comp = FactoryBot.create(:competition, :confirmed, delegates: [delegate], registration_requirements: "Requirements")
        new_requirements = "New requirements"
        patch :update, params: { id: comp, competition: { registration_requirements: new_requirements } }
        comp.reload
        expect(comp.registration_requirements).to eq "Requirements"
````

I think that makes it clearer to the test reader what's going on, by bringing all the "relevant" information for the test into the test.